test/celeritas/optical/Rayleigh.test.cc

```diff
@@ -0,0 +1,140 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/OpticalRayleigh.test.cc
+//---------------------------------------------------------------------------//
+#include "celeritas/optical/model/RayleighInteractor.hh"
+#include "celeritas/phys/InteractorHostTestBase.hh"
+
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+namespace test
+{
+using namespace ::celeritas::test;
+
+//---------------------------------------------------------------------------//
+
+class RayleighInteractorTest : public InteractorHostTestBase
+{
+  protected:
+    void SetUp() override
+    {
+        inc_direction_ = {0, 0, 1};
+
+        // Check incident quantities are valid
+        sanity_check(inc_direction_, track_view().polarization());
+    }
+
+    void sanity_check(Real3 const& direction, Real3 const& polarization) const
+    {
+        // Check that direction and polarization are unit vectors
+        EXPECT_SOFT_EQ(1, norm(direction));
+        EXPECT_SOFT_EQ(1, norm(polarization));
+        // Check that direction and polarization are perpendicular
+        EXPECT_SOFT_EQ(0, dot_product(direction, polarization));
+    }
+
+    void sanity_check(Interaction const& interaction) const
+    {
+        // Interactions should always be scattering
+        EXPECT_EQ(Interaction::Action::scattered, interaction.action);
+        sanity_check(interaction.direction, interaction.polarization);
+    }
+
+    TrackView const& track_view() const
+    {
+        // Energy doesn't matter; initial polarization should be perpendicular
+        static TrackView view{units::MevEnergy{13e-6}, {1, 0, 0}};
+        return view;
+    }
+
+    Real3 inc_direction_;
+};
+
+TEST_F(RayleighInteractorTest, basic)
+{
+    int const num_samples = 4;
+
+    RayleighInteractor interact{track_view(), inc_direction_};
+
+    auto& rng_engine = this->rng();
+
+    std::vector<real_type> dir_angle;
+    std::vector<real_type> pol_angle;
+
+    for ([[maybe_unused]] int i : range(num_samples))
+    {
+        Interaction result = interact(rng_engine);
+        this->sanity_check(result);
+
+        dir_angle.push_back(dot_product(result.direction, inc_direction_));
+        pol_angle.push_back(
+            dot_product(result.polarization, track_view().polarization()));
```

Comment:
A good additional test is the product of the polarization vector and the direction.

Reply:
I overlooked the sanity check, which may also include  `dot_product(final direction, final polarization) = 0`